### ISSUE TITLE
Feature Request/Question: Recuring payments and Custom Gateways

### ISSUE THREAD

**Issue by kram-dyoll** [reachweb/statamic-resrv#132]
Hello

First off, this booking system looks awesome. I'm thinking of using this on a clients project but they have some requirements that this addon does not seem to provide out of the box. 

1. I assume adding a custom payment gateway will not be an issue as I can see the interfaces used and where to set which Gateway to use. Is my assumption correct?
2. The client offers childcare where places can be booked for the year and paid monthly. This looks like something that cannot be setup with Resrv, so I was wondering if this is on a roadmap? 
3. Regarding recuring payments. The client is currently setup for regular bank payments with a team that can administer this. However this would require Resrv to book out multiple availabilities (which looking at the documentation, I think this can be done), but the payment will be 0 as this will be handled by the admin team.
4. I will need to build a custom gateway. Would I be able to add a recurring payment option using standard Statamic fields for the checkout or would this be out of scope for Resrv in it's current form?

This is of course not what Resrv is designed for, but could Resrv be used for this kind of system out of the box, or will changes be required?

Thanks

**Comment by afonic** [comment]
Hello,

I'm glad you like Resrv, thank you!

1. Building a custom payment gateway is pretty straight forward if you follow the interface. You can switch it using the config file as we do for Mollie: https://resrv.dev/mollie#switch-the-payment-gateway . We've build more than just Stripe and Mollie if it's a popular one let me know and I can add it to the list.
2. There are no recurring payments support right now. I'd like to tell you that I'll add it but to be honest it doesn't really make sense for a reservation system. Let me think about it. However what you could do is set each reservation to require a fixed amount or percentage in the config (as a down payment - check https://resrv.dev/configuration#checkout-settings) and then in your custom Payment gateway make this payment recurring.
3. Yes you block the availability only and if the price is zero Resrv just skips the payment step. So you can setup availability and pricing, display the price and apply a 100% discount Dynamic Pricing to skip the payment step.
4. Right now there is no way to switch the payment method (or the payment gateway) based on the Collection or Entry. Having said that, we have a client request to allow multiple payment methods for each reservations so I guess we will soon considerably rewrite that part of the add-on.

To sum up I think Resrv can be configured to be close to what you need. It certainly can be used to handle availability only and bypass payment and depending on your use case it might also accept recurring payments.

Let me know if you need more help, you can also find me in Statamic's Discord server (username Iosif).

**Comment by kram-dyoll** [comment]
Thank you very much for your swift and detailed reply. This answers my questions and will allow me to investigate further.

I'm not sure what payment gateway we will use, but where I'm based we cannot use Stripe or Mollie.

I'm happy to have this issue closed if you would like to keep things tidy.
